### DISCUSSION TITLE
SeaweedFS Appreciation Post

### DISCUSSION THREAD

**Discussion by Material-Scientist** [seaweedfs/seaweedfs#5927] [Show and tell]
Just a post to appreciate how SeaweedFS + S3 API simplifies secure data sharing ❤️ 

Can now do this:
```python
import pandas as pd
import fsspec # abstracts s3 api

s3_options = {
    'key': 'key',
    'secret': 'secret',
    'client_kwargs': {
        'endpoint_url': 'https://s3.my-page.com',
    }
}

uri = 's3://default/dummy_file.csv'

df = pd.read_csv(
    uri,
    storage_options=s3_options
)
df
```
![image](https://github.com/user-attachments/assets/4e7bc8b6-14f0-4b97-8716-e332fde2296b)

Can also use `pd.to_csv` (or `to_parquet`) to write files to SeaweedFS via S3 API:
```python
uri2 = 's3://default/my_output_file.csv'

df.to_csv(
    uri2,
    storage_options=s3_options
)
```
![image](https://github.com/user-attachments/assets/032609bb-f9ab-4b81-811f-ef74b08c9087)

This becomes extremely useful when you use e.g. [Coiled](https://www.coiled.io/) to spin up temporary [Dask](https://docs.dask.org/en/stable/) workers that can accelerate your data processing pipeline 100-fold!

Thanks @chrislusf & maintainers!

**Comment by chrislusf** [comment]
This is great! Thanks! 

It's great to know how you use it. Coild and Dask are interesting and enlightening.
